tools/codegen/api/python.py

```diff
@@ -173,6 +173,53 @@
 #      }
 #
 
+# TODO: stick this more firmly in the data model somewhere?
+def namedtuple_fieldnames(returns: Tuple[Return]) -> List[str]:
+    if len(returns) <= 1 or all(map(lambda r: r.name is None, returns)):
+        return []
+    else:
+        if any(map(lambda r: r.name is None, returns)):
+            # When building on Windows, `PyStructSequence_UnnamedField` could not be
+            # resolved by the linker for some reason, which cause error in building:
+            #
+            # python_nn_functions.cpp.obj : error LNK2001: unresolved external symbol
+            # PyStructSequence_UnnamedField
+            #
+            # Thus, at this point in time, we do not support unnamed
+            # fields in namedtuple; you must either name all fields,
+            # or none of them.
+            raise ValueError("Unnamed field is not supported by codegen")
+
+        return list(map(lambda r: str(r.name), returns))
```

Comment:
I didn't see an update to tools/autograd/gen_python_functions.py in this PR so there are still two copies of this function?